digicampipe/calib/camera/random_triggers.py

```diff
@@ -2,51 +2,25 @@
 
 
 def fill_baseline_r0(event_stream, n_bins=10000):
-
-    count_calib_events = 0
-
-    for count, event in enumerate(event_stream):
-
+    n_events = None
+    baselines = []
+    baselines_std = []
+    for event in event_stream:
         for telescope_id in event.r0.tels_with_data:
-
-            if count == 0:
-                n_pixels = event.inst.num_pixels[telescope_id]
-                n_samples = event.inst.num_samples[telescope_id]
-                n_events = n_bins // n_samples
-                baselines = np.zeros((n_pixels, n_events))
-                baselines_std = np.zeros((n_pixels, n_events))
-                baseline = np.zeros(n_pixels)
-                std = np.zeros(n_pixels)
-
             r0_camera = event.r0.tel[telescope_id]
+            adc_samples = r0_camera.adc_samples
+            if n_events is None:
+                n_events = n_bins // adc_samples.shape[1]
 
             if r0_camera.camera_event_type == 8:
-
-                adc_samples = r0_camera.adc_samples
-                new_mean = np.mean(adc_samples, axis=-1)
-                new_std = np.std(adc_samples, axis=-1)
-
-                baselines = np.roll(baselines, 1, axis=-1)
-                baselines_std = np.roll(baselines_std, 1, axis=-1)
-
-                baseline += new_mean - baselines[..., 0]
-                std += new_std - baselines_std[..., 0]
-
-                baselines[..., 0] = new_mean
-                baselines_std[..., 0] = new_std
-
-                count_calib_events += 1
-
-            if count_calib_events >= n_events:
-
-                r0_camera.baseline = baseline / n_events
-                r0_camera.standard_deviation = std / n_events
-
-            else:
-
-                r0_camera.baseline = np.zeros(n_pixels) * np.nan
-                r0_camera.standard_deviation = np.zeros(n_pixels) * np.nan
-
+                baselines.append(np.mean(adc_samples, axis=1))
```

Comment:
see next line below this

Reply:
In principle one should use a [`collections.deque`](https://docs.python.org/3.6/library/collections.html#deque-objects) here, since this has a maximum length and appending a new event will automagically drop the oldest event. But I've read somewhere that it is slow .. and did not even try it out. 

Using a deque here would probably save us another two lines of code...

Reply:
Nope I was wrong. A deque seems to be a factor 3 faster...
```
In [7]: %%timeit
   ...: d = deque(maxlen=10)
   ...: for i in range(100):
   ...:     d.append(i)
5.73 µs ± 35.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [8]: %%timeit
   ...: l = []
   ...: for i in range(100):
   ...:     l.append(i)
   ...:     l = l[-10:]
16.9 µs ± 131 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
I'll have to test and see if `deque` plays nicely with numpy, so is: 

`numpy.mean(some_deque_with_arrays_inside)`

 as fas as 

`numpy.mean(some_list_with_arrays_inside)`

If this is equally fast I will use a deque and save some lines and introduce the reader of this code to `collections.deque`, which is maybe a good thing.

Reply:
Great I am learning new stuff. I was sure this data structure existed. 
I also think having a deque makes clear what we are doing here

Reply:
FYI : https://pypi.python.org/pypi/numpy_ringbuffer/0.2.0

Reply:
Oh .. such a ringbuffer might be even better/faster, while not making the code more difficult to read.
I'll have a look. Is an additional external dependency fine for us?